public/anomaly_detection_app.tsx

```diff
@@ -29,13 +34,18 @@ export function renderApp(coreStart: CoreStart, params: AppMountParameters) {
   } else {
     require('@elastic/charts/dist/theme_only_light.css');
   }
+
   ReactDOM.render(
     <Provider store={store}>
       <Router>
         <Route
           render={(props) => (
             <CoreServicesContext.Provider value={coreStart}>
-              <Main {...props} />
+              <Main
+                dataSourceEnabled={dataSource.dataSourceEnabled}
+                setHeaderActionMenu={params.setHeaderActionMenu}
```

Comment:
1. Can we fetch `dataSourceEnabled` from services? `getDataSourceManagementPlugin().dataSourceEnabled`. Then we can avoid passing extra props
2. Rather than passing down the `setHeaderActionMenu` to all components which adds lots of extra props, can we persist it on initialization? Maybe exploring adding this in redux store somewhere, and can be fetched there. `configureStore` is exposed here to instantiate the client, this may be a good place where that could be added

Reply:
Having props passed down isn't inherently bad. I do agree that too many props being passed down multiple levels might lead to prop drilling, which can make the code harder to understand and maintain. However, in this case, we're not passing lots of extra props, just this one setHeaderActionMenu function as a prop. It's pretty straightforward as it's not being passed down multiple levels, just passed down one level to the route path landing page. So, it doesn't make much difference performance-wise and maintenance-wise I believe. Also, I'm not sure how to add this param to the Redux configureStore, like create a new Redux store for this? Do you have an example of how to do it?

Reply:
1. Thanks!
2. Got it, since this is already configured in the `AppMountParameters` and will only go one level down, that can be ok.